app/src/main/kotlin/com/rockspin/bargainbits/ui/deals/DealListActivity.kt

```diff
@@ -0,0 +1,136 @@
+package com.rockspin.bargainbits.ui.deals
+
+import android.app.SearchManager
+import android.arch.lifecycle.ViewModelProviders
+import android.content.Context
+import android.content.Intent
+import android.databinding.DataBindingUtil
+import android.os.Bundle
+import android.support.design.widget.Snackbar
+import android.support.v4.view.MenuItemCompat
+import android.support.v7.widget.SearchView
+import com.jakewharton.rxbinding2.support.v7.widget.itemClicks
+import com.jakewharton.rxbinding2.widget.itemSelections
+import com.rockspin.bargainbits.R
+import com.rockspin.bargainbits.databinding.ActivityDealListBinding
+import com.rockspin.bargainbits.ui.BaseActivity
+import com.rockspin.bargainbits.ui.store_filter.StoreFilterDialogFragment
+import com.rockspin.bargainbits.ui.watch_list.WatchListActivity
+import com.rockspin.bargainbits.util.visible
+import com.rockspin.bargainbits.utils.Feedback
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.rxkotlin.addTo
+import io.reactivex.rxkotlin.merge
+import io.reactivex.subjects.PublishSubject
+import org.codechimp.apprater.AppRater
+import javax.inject.Inject
+
+class DealListActivity : BaseActivity(), StoreFilterDialogFragment.OnDialogClosedListener {
+
+    @Inject lateinit var adapter: DealListAdapter
+    @Inject lateinit var factory: DealListViewModelFactory
+
+    private var noInternetSnackbar: Snackbar? = null
+    private val onDestroyDisposable = CompositeDisposable()
+    private val onStoreFilterClosed = PublishSubject.create<Unit>()
+
+    private val binding by lazy { DataBindingUtil.setContentView<ActivityDealListBinding>(this, R.layout.activity_deal_list) }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // TODO - remember user default choice for deal sorting
+
+        binding.toolbar.inflateMenu(R.menu.deal_list)
+
+        val searchManager = getSystemService(Context.SEARCH_SERVICE) as SearchManager
+        val searchView = MenuItemCompat.getActionView(binding.toolbar.menu.findItem(R.id.menu_search_games)) as SearchView
+        searchView.setSearchableInfo(searchManager.getSearchableInfo(componentName))
+
+        binding.dealsRecyclerView.adapter = adapter
+
+        val viewModel = ViewModelProviders.of(this, factory).get(DealListViewModel::class.java)
+
+        listOf(
+            binding.dealSortTypeSpinner.itemSelections().map { DealListEvent.SortingChanged(it) }.skip(1),
```

Comment:
I'm still trying to tie down how this architecture ideally looks. Could you please ignore this part for now and keep your notes in mind for further PRs? The comments are valid but I'd like to have another review once i've cleaned this up.